src/Fabulous/Core.fs

```diff
@@ -88,21 +91,29 @@ and [<Struct>] WidgetDiff =
         WidgetChanges: WidgetChange []
         WidgetCollectionChanges: WidgetCollectionChange []
     }
-    
+
 /// Context of the whole view tree
-type [<Struct>] ViewTreeContext =
-    { CanReuseView: Widget -> Widget -> bool
-      GetViewNode: obj -> IViewNode
-      Dispatch: obj -> unit }
-    
+[<Struct>]
+type ViewTreeContext =
+    {
+        CanReuseView: Widget -> Widget -> bool
+        GetViewNode: obj -> IViewNode
+        Dispatch: obj -> unit
+    }
+
 and IViewNode =
     abstract member Target : obj
     abstract member Parent : IViewNode voption
     abstract member TreeContext : ViewTreeContext
     abstract member MapMsg : (obj -> obj) voption with get, set
-    abstract member GetViewNodeForChild: obj -> IViewNode
+
+    // whatever we need to attach to ViewNode in Core
+    // TODO if it is for internal use only it is possible that we should strongly type
```

Comment:
Makes sense, I actually wonder if we can get perf boost to avoid using IViewNode, and use ViewNode directly instead. I think we found a way to use Node directly in a pretty universal way, but I'm fuzzy on details there.

If you don't mind I would prefer to merge it as is, I have an intuition that we will find a good solution there eventually

Reply:
I tried to avoid as many interface as possible from my last big refactor, but IViewNode is the only one that remained.
This one is annoying to get rid of because of recursive use between AttributeDefinition and ViewNode, it would mean to put them in a same file.

Sure, we can keep it like that for now.